packages/selected-network-controller/tests/SelectedNetworkController.test.ts

```diff
@@ -415,76 +415,119 @@ describe('SelectedNetworkController', () => {
+        const mockProxyProvider = {
+          setTarget: jest.fn(),
+        } as unknown as ProviderProxy;
+        const mockProxyBlockTracker = {
+          setTarget: jest.fn(),
+        } as unknown as BlockTrackerProxy;
 
-      it('throws an error if the domain network client is not initialized', () => {
-        const { controller, mockGetNetworkClientById } = setup({
-          state: {
-            domains: {
-              'example.com': 'mainnet',
+        const domainProxyMap = new Map<Domain, NetworkProxy>([
+          [
+            'example.com',
+            {
+              provider: mockProxyProvider,
+              blockTracker: mockProxyBlockTracker,
             },
-          },
-          useRequestQueuePreference: true,
-        });
-        mockGetNetworkClientById.mockImplementation(() => {
-          throw new Error('No network client was found with the ID');
-        });
-
-        expect(() =>
-          controller.getProviderAndBlockTracker('example.com'),
-        ).toThrow('No network client was found with the ID');
-      });
-    });
-
-    describe('provider and block tracker are not cached for the passed domain and networkClientId has not been set for the requesting domain', () => {
-      it('creates a new proxy provider and block tracker with the global network client', () => {
+          ],
+          [
+            'test.com',
+            {
+              provider: mockProxyProvider,
+              blockTracker: mockProxyBlockTracker,
+            },
+          ],
+        ]);
         const { controller } = setup({
           state: {
             domains: {},
           },
           useRequestQueuePreference: true,
+          domainProxyMap,
         });
-        const result = controller.getProviderAndBlockTracker('test.com');
-        expect(result).toBeDefined();
-      });
 
-      it('throws an error if the global network client is not initialized', () => {
-        const { controller, mockGetSelectedNetworkClient } = setup({
-          state: {
-            domains: {},
-          },
-          useRequestQueuePreference: false,
+        const result = controller.getProviderAndBlockTracker('example.com');
+        expect(result).toStrictEqual({
+          provider: mockProxyProvider,
+          blockTracker: mockProxyBlockTracker,
         });
-        mockGetSelectedNetworkClient.mockReturnValue(undefined);
-        expect(() => controller.getProviderAndBlockTracker('test.com')).toThrow(
-          'Selected network not initialized',
-        );
       });
     });
+    describe('when the domain does not have a cached networkProxy in the domainProxyMap', () => {
+      describe('when the useRequestQueue preference is true', () => {
+        describe('when the domain has permissions', () => {
+          it('calls to NetworkController:getNetworkClientById and creates a new proxy provider and block tracker with the global network client', () => {
```

Comment:
Would it be more clear if these scenarios say something like 

`creates a new proxy provider and block tracker with the non-proxied global network client`
or
`proxied global network client`? idk